### PR TITLE
Support conditional requirements in old versions of pip

### PR DESCRIPTION
**Pull request by graingert**
idea from https://github.com/jkbrzt/httpie/blob/master/setup.py#L46

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -20,6 +21,21 @@ def run_tests(self):
...
+
+
+extras_require = {
+    ':python_version in "2.4, 2.5, 2.6"':
+        ['contextlib2', 'backport_collections', 'mock'],
+    ':python_version in "2.7, 3.1, 3.2"': ['contextlib2', 'mock'],
+}
+
+
+if 'bdist_wheel' not in sys.argv:
```
Comment (IvanMalison) [inline setup.py]: can we not do a version check here too. Won't this make it do it unconditionally?

**Comment by graingert** [comment]
@IvanMalison I see you made a comment, (I got an email notification) but it's gone now...

**Comment by IvanMalison** [comment]
Yeah i was worried that we would do things unconditionally, but I was just misreading the code. Looks good to me. I wish there were a way we could test.

**Comment by graingert** [comment]
I did a:
```
pip install pip==1.5
pip install .
```
in my local checkout

**Comment by IvanMalison** [comment]
yeah I guess we could add a travis environment that does the same thing?

**Comment by IvanMalison** [comment]
closes #169

**Comment by kevin1024** [comment]
Nice! Thanks @graingert.

**Pull request merged**